src/nnfusion/engine/pass/graph/runtime_const_folding_pass.hpp

```diff
@@ -17,16 +22,47 @@ namespace nnfusion
         {
             class RuntimeConstantFoldingPass : public GraphPassBase
             {
-                int runtime_const_folding_iterate_once(
-                    std::shared_ptr<Graph>& graph,
-                    std::set<std::shared_ptr<GNode>>& blocklist_nodes);
+                std::shared_ptr<GNode>
+                    runtime_const_folding_node(std::shared_ptr<Graph>& graph,
+                                               std::set<std::shared_ptr<GNode>>& blocklist_nodes,
+                                               std::shared_ptr<GNode>& node);
+                void runtime_const_folding_task(std::shared_ptr<Graph>& graph,
+                                                std::set<std::shared_ptr<GNode>>& blocklist_nodes,
+                                                std::shared_ptr<GNode> node,
```

Comment:
>std::shared_ptr<GNode> node, [](http://example.com/codeflow?start=48&length=28)

💡[Discussion]

I see that most of the shared_ptrs are passed as reference, but for this one, it is passed by value. Why? And besides, I have questions about if we should pass these shared pointers by reference or value. You may take some time to make sure your choice is right.

Reply:
Here is a reference: [https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/](https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/)

Reply:
Use value here was because the node may be changed if it isn't a const node. (in the impl file line 297 use `node = runtime_const_folding_node(graph, blocklist_nodes, node);`)
After your comment, I realize we don't need the reference of shared pointers here, so I change them to values.